### ISSUE TITLE
Any plans to support metrics20.org spec?

### ISSUE THREAD

**Issue by petemounce** [Recognos/Metrics.NET#26]
http://metrics20.org/spec/ + http://dieter.plaetinck.be/post/monitorama-pdx-metrics20/

The benefits from the canonical list of keys include:
* easier to define checks, queries, and dashboards because of metric naming consistency
* easier to write tools against the more consistent way of reasoning about metrics
* easier to define rollup patterns, and more flexible, since one can decide to keep particular dimensions (like cloud server id) for a short time (since they're more ephemeral and only relevant short term), and other dimensions for a longer time

[Netflix's Atlas monitoring tool](http://techblog.netflix.com/2014/12/introducing-atlas-netflixs-primary.html) independently seems to arrived at the same sort of conclusions as the spec's author.

**Comment by alhardy** [comment]
Didnt know there was some type of spec, this is cool. Who's defined that spec? Is it adopted at all?

**Comment by petemounce** [comment]
I think @obfuscurity defined it. Like I said, the good people at Netflix independently arrived at something very similar with their Atlas monitoring stack.

There's support in [some implementations](http://metrics20.org/implementations/) of both metrics aggregators and visualizers.

At just eat, we're using a similar idea for logs, but not yet metrics, and are laying the groundwork to adopt the spec. If you're interested in specs along these lines, http://opentracing.io/ is another one to be aware of.

**Comment by petemounce** [comment]
Pardon me, it was @Dieterbe.

**Comment by Dieterbe** [comment]
yup, me. Hi. the response to adption question is quite interesting. lots of people have reached out and said they're interested in adopting and "like the ideas in the spec". However almost all the listed implementations are my doing and i only know of 1 other person who adopted it in his project.  So I think there is some friction somewhere that makes metrics2.0 hard to adopt. I actually recently gave it a big overhaul (see the last news entry http://metrics20.org/news/), so i would be curious to hear what you folks think, if anything is confusing or why it may be hard to adopt.
